deps/src/singular.cpp

```diff
@@ -53,6 +72,45 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     singular_define_ideals(Singular);
     singular_define_matrices(Singular);
 
+    Singular.method("call_interpreter", [](std::string input) {
+        // set callbacks
+        auto default_print = PrintS_callback;
+        auto default_error = WerrorS_callback;
+        auto default_warning = WarnS_callback;
+
+        PrintS_callback = PrintS_for_julia;
+        WerrorS_callback = WerrorS_for_julia;
+        WarnS_callback = WarningS_for_julia;
+
+        // cleanup return strings
+        singular_return.clear();
+        singular_error.clear();
+        singular_warning.clear();
+
+        // call interpreter
+        std::string  input_str = input + "\nreturn();";
+        char *       ost = (char *)omalloc0(input_str.length() + 1);
+        memcpy(ost, input_str.c_str(), input_str.length() + 1);
+        bool err = iiAllStart(NULL, ost, BT_proc, 0);
+        inerror = 0;
+        errorreported = 0;
+        omFree(ost);
+
+        // Get output
+        jl_array_t * result = jl_alloc_array_1d(jl_array_any_type, 4);
+        jl_arrayset(result, err ? jl_true : jl_false, 0);
+        jl_arrayset(result, jl_cstr_to_string(singular_return.c_str()), 1);
+        jl_arrayset(result, jl_cstr_to_string(singular_error.c_str()), 2);
+        jl_arrayset(result, jl_cstr_to_string(singular_warning.c_str()), 3);
+
+        // reset handlers 
```

Comment:
You write "callbacks" above and then "handlers" here. Pick one and stick to it? Also, perhaps say "restore" instead of "reset?